src/Streamly/Mem/Array.hs

```diff
@@ -321,27 +385,57 @@ readIndex arr i =
     else liftIO $ withForeignPtr (aStart arr) $ \p ->
             fmap Just $ peekElemOff p i
 
-{-
+
 -- | @readSlice arr i count@ streams a slice of the array @arr@ starting
 -- at index @i@ and reading up to @count@ elements in the forward direction
 -- ending at the index @i + count - 1@.
 --
+-- >>> S.toList $ readSlice (fromList [1..10]) 3 4
+-- > [4,5,6,7]
+--
 -- @since 0.7.0
 {-# INLINE readSlice #-}
```

Comment:
Make it `INLINE_NORMAL` so that it can be used efficiently in higher order functions.